activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -65,6 +66,20 @@ def validate_each(record, attr_name, value, precision: Float::DIG, scale: nil)
       end
 
     private
+      def option_as_range(record, attr_name, option_value)
```

Comment:
Database specific logic (`columns_hash`) cannot go in Active Model, it should be in the Active Record numericality validator.